src/main/java/com/intellij/rt/debugger/agent/CaptureStorage.java

```diff
@@ -37,6 +40,23 @@ private static Deque<CapturedStack> getStacksForCurrentThread() {
     }
   }
 
+  private static void cleanUpThreadToStackMap() {
```

Comment:
We could also try doing smth less dumb, e.g.
* not traversing all the elements every time we clean-up, but log2(n)
* or maybe adjust THRESHOLD_MULTIPLIER dynamically